lightning/src/ln/channel.rs

```diff
@@ -1684,6 +1685,103 @@ pub(super) trait InteractivelyFunded<SP: Deref> where SP::Target: SignerProvider
 
 	fn dual_funding_context(&self) -> &DualFundingChannelContext;
 
+	fn dual_funding_context_mut(&mut self) -> &mut DualFundingChannelContext;
+
+	fn is_initiator(&self) -> bool;
+
+	#[allow(dead_code)] // TODO(dual_funding): Remove once contribution to V2 channels is enabled
+	fn begin_interactive_funding_tx_construction<ES: Deref>(
+		&mut self, signer_provider: &SP, entropy_source: &ES, holder_node_id: PublicKey,
+		extra_input: Option<(TxIn, TransactionU16LenLimited)>,
```

Comment:
Oh I missed this one, sorry! It's not entirely clear to me what this `extra_input` field refers to since the method isn't used yet. I take it it's not the shared input, right?

Reply:
lightning/src/ln/channel.rs

Reply:
It's used by splicing, for automatically adding the current funding transaction as an input. You can see the usage in #3444 here:
https://github.com/lightningdevkit/rust-lightning/pull/3444/files#diff-8d28e96b6f27edb9a9739b2dce334f8906ec906155d421a7b5b02aa8e6c96057

`begin_interactive_funding_tx_construction` is called twice, once with `None` (acceptor side), and once with the current funding transaction.

If you prefer, I can take it out from this PR, and add it only in the splicing PR, since this is splicing specific.

Reply:
I think it's fine to have it here but maybe with docs and maybe we should just call it `prev_funding_input`, what do you think?